spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java

```diff
@@ -123,7 +124,7 @@ public MimeMessage send(final @NonNull Email email,
                 DataSource source = new ByteArrayDataSource(emailAttachment.getAttachmentData(),
                         emailAttachment.getContentType().toString());
                 attachmentPart.setDataHandler(new DataHandler(source));
-                attachmentPart.setFileName(emailAttachment.getAttachmentName());
+                attachmentPart.setFileName(MimeUtility.encodeText(emailAttachment.getAttachmentName()));
```

Comment:
For my project, this is the difference between having a file name or not having a file name. Unfortunately I cannot explain why :(
If you look at Spring MimeMessageHelper line 996, you will see that they do encode attachments file name, using default system encoding. You are using this code for emails not based on templates.
You are probably right about using the email encoding, I guess MimeMessageHelper.addAttachment is not the best piece of code in Spring ;)